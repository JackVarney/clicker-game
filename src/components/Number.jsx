import React from "react";
import styled from "styled-components";

const NumberText = styled.span`
  display: block;
`;

function Number({ children, label }) {
  return <NumberText>{`${label ? label + ": " : ""}${children}`}</NumberText>;
}

export default Number;
