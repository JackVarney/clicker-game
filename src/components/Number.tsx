import React, { FC } from "react";
import styled from "styled-components";

const NumberText = styled.span`
  display: block;
`;

interface Props {
  label?: string;
}

const Number: FC<Props> = ({ children, label }) => (
  <NumberText>{`${label ? label + ": " : ""}${children}`}</NumberText>
);

export default Number;
