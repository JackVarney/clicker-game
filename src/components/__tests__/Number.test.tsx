import "jest-styled-components";
import React from "react";
import renderer from "react-test-renderer";
import Number from "../Number";

it("it should render label when label is passed", () => {
  const tree = renderer.create(<Number label={"label"}>{0}</Number>).toJSON();

  expect(tree).toMatchSnapshot();
});

it("should render no label when no label passed", () => {
  const tree = renderer.create(<Number>{0}</Number>).toJSON();

  expect(tree).toMatchSnapshot();
});
