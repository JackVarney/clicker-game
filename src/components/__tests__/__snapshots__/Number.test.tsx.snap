// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render label when label is passed 1`] = `
.c0 {
  display: block;
}

<span
  className="c0"
>
  label: 0
</span>
`;

exports[`should render no label when no label passed 1`] = `
.c0 {
  display: block;
}

<span
  className="c0"
>
  0
</span>
`;
